calculator.c

```diff
@@ -0,0 +1,431 @@
+//
+//  calculator.c
+//  Calculator
+//
+//  Created by Jonas on 25/01/2020.
+//  Copyright � 2020 Jonas. All rights reserved.
+//
+
+#include "calculator.h"
+#include "calculatorTest.h"
+
+
+
+/**
+ *  This fuction takes a long long number and checks
+ *  if it is bigger than INT_MAX, that is, it checks
+ *  if result can be contained in an int without overflowing
+ *  it returns its input as an int if result can be contained in
+ *  an int witout overflowing, else it returns -1 that will never be taken
+ *  into account because flag NO_ERROR will be assigned to 1.
+ *  @param long result, the numer to be checked.
+ *  @return the input as an ,int or -1 and raise flag NO_ERROR.
+ */
+int checkOverflowAndReturnResult(long long result) {
+	if (result > INT_MAX) {
+		ERROR_CODE = intOverflow;
+		return -1;//doesnt matter
+	}
+	return (int)result;
+}
+
+/**
+ *  This fuction takes two int as input, calculate their sum as a
+ *  long long and call the fucntion checkOverflowAndReturnResult
+ *  that will return the input as int if the result does not int overflow,
+ *  or it will return -1 and raise flag NO_ERROR so that
+ *  the result will not be taked into account.
+ *  @param int firstInt, int secondInt, the numbers to be added.
+ *  @return the sum of the numbers as int, or -1 and raise flag NO_ERROR.
+ */
+int add(int firstInt, int secondInt) {
+	//pointer to fucntion declaration
+	int (*ptrCheckOverflowAndReturnResult)(long long) = &checkOverflowAndReturnResult;
+	long long result = (long long)firstInt + (long long)secondInt;
+	return (*ptrCheckOverflowAndReturnResult)(result);
+}
+
+/**
+ *  This fuction takes two int as input, calculate their substraction
+ *  and return the result, note that in our case, substract
+ *  cannot int overflow because it takes as input two positive ints < INT_MAX
+ *  @param int firstInt, int secondInt, the numbers to be substracted.
+ *  @return firstInt - secondInt, the substraction as int.
+ */
+int substract(int firstInt, int secondInt) {
+	return firstInt - secondInt;

```

Comment:
Hey :)
In the code, I think it can't as we don't deal with negative numbers, 
But I can still add a verification for underflow in case we decide (in the future) to support negative numbers

Reply:
So what happens if I calculate "`3-5`"?

Reply:
it will work, but we cant have an underflow (I think) as we catn get input
a = -INT_MAX+1
b = INT_MAX-1
i this case a-b will product underflow  but we dont support here this case
in our case if a,b<INT_MAX then -INT_MAX<a-b<INT_MAX
pls correct me if im wong :)